src/index.js

```diff
@@ -175,6 +171,39 @@ async function check(autoCheckElement: AutoCheckElement) {
   }
 }
 
+async function processResponse(response: Response, input: HTMLInputElement, required: boolean) {
```

Comment:
\~This is just a feeling\~:

I expected more calls in this big block to be async, but in reality there's just one, and might not even be called if `required === false`. 

What do you think about separating these out into a few calls from this function?